keyboards/planck/keymaps/gunp/keymap.c

```diff
@@ -27,136 +26,173 @@ enum planck_layers {
   LY_0100, LY_0101, LY_0110, LY_0111,
   LY_1000, LY_1001, LY_1010, LY_1011,
   LY_1100, LY_1101, LY_1110, LY_1111,
+  LY_SANDBOX, LY_RELAX, LY_THAI_A, LY_THAI_B, LY_THAI_C,
 };
 
 enum planck_keycodes {
   FN_A = SAFE_RANGE,
   FN_B,
   FN_C,
   FN_D,
+  SANDBOX,
+  RELAX,
+  THAI_A,
+  THAI_B,
+  THAI_C,
+  RESET_CONF,
+  OS_LINUX,
+  OS_WIN,
+  OS_OSX,
   DYNAMIC_MACRO_RANGE,
 };
 
 #include "dynamic_macro.h"
 
-#define MFN_R1 XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
-#define MFN_R2 XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
-#define MFN_R3 XXXXXXX, FN_A,    FN_B,    FN_C,    XXXXXXX, XXXXXXX
-#define MFN_R4 XXXXXXX, KC_LCTL, KC_LSFT, KC_LALT, FN_D,    XXXXXXX
+#define MFN_R1 XXXXXXX, KC_LEFT, KC_BSPC, KC_RGHT, KC_SPC,  KC_TAB
+#define MFN_R2 XXXXXXX, THAI_A,  THAI_B,  THAI_C,  XXXXXXX, KC_ENT
+#define MFN_R3 XXXXXXX, FN_A,    FN_B,    FN_C,    FN_D,    KC_LSFT
+#define MFN_R4 RELAX, RESET_CONF, KC_APP, KC_LGUI, KC_LALT, KC_LCTL
+
 #define UNUSED_LAYER {\
   {XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MFN_R1},\
   {XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MFN_R2},\
   {XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MFN_R3},\
   {XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MFN_R4}},
-#define MC(kc) LCTL(kc)
-#define MA(kc) LALT(kc)
-#define MG(kc) LGUI(kc)          // ModGui, for i3 shortcuts
+
+#define F1_F12_LAYER(M){\
+ {S(M(KC_F7 )),S(M(KC_F8 )),M(KC_F7 ),M(KC_F8 ),M(KC_F9 ),S(M(KC_F9 )),MFN_R1},\
+ {S(M(KC_F4 )),S(M(KC_F5 )),M(KC_F4 ),M(KC_F5 ),M(KC_F6 ),S(M(KC_F6 )),MFN_R2},\
+ {S(M(KC_F1 )),S(M(KC_F2 )),M(KC_F1 ),M(KC_F2 ),M(KC_F3 ),S(M(KC_F3 )),MFN_R3},\
+ {S(M(KC_F10)),S(M(KC_F11)),M(KC_F10),M(KC_F11),M(KC_F12),S(M(KC_F12)),MFN_R4}},
+
+#define MC(kc)  LCTL(kc)
+#define MA(kc)  LALT(kc)
+#define MG(kc)  LGUI(kc)
 #define MSC(kc) LSFT(LCTL(kc))
-#define MSA(kc) LSFT(LALT(kc))
-#define MCA(kc) LCTL(LALT(kc))
-#define XX XXXXXXX
+
+#define XXX XXXXXXX
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 [LY_0000] = {
```

Comment:
It may be a good idea to update things to use the `LAYOUT_ortho_4x12` macro rather than the raw matrix, especially if you ever plan on getting a rev6 or EZ, as they don't use the same matrix that the AVR boards use.

Reply:
sorry, but I don't know where to put `LAYOUT_ortho_4x12` in. do you have the tutorial?